File: benchmarks/go/mixed.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	a := make([]int, 10000)
	for i := range a { a[i] = i }
	b := make([]int, len(a))
	for i, v := range a { b[i] = v * 2 }
	var c []int
	for _, v := range b {
		if v%2 == 0 { c = append(c, v) }
	}
	sum := 0
	for _, v := range c { sum += v }

	obj := map[string]int{"sum": sum, "count": len(c)}
	data, _ := json.Marshal(obj)
	os.WriteFile("/tmp/sans_bench_mixed.txt", data, 0644)
	raw, _ := os.ReadFile("/tmp/sans_bench_mixed.txt")
	var parsed map[string]int
	json.Unmarshal(raw, &parsed)
	fmt.Println(parsed["sum"])
}

File: benchmarks/rust/mixed.rs
use std::fs;
use std::collections::HashMap;

fn main() {
    let a: Vec<i64> = (0..10000).collect();
    let b: Vec<i64> = a.iter().map(|&x| x * 2).collect();
    let c: Vec<i64> = b.iter().filter(|&&x| x % 2 == 0).cloned().collect();
    let sum: i64 = c.iter().sum();

    // Simple JSON serialization without deps
    let s = format!("{{\"sum\":{},\"count\":{}}}", sum, c.len());
    fs::write("/tmp/sans_bench_mixed.txt", &s).unwrap();
    let raw = fs::read_to_string("/tmp/sans_bench_mixed.txt").unwrap();
    // Parse sum from JSON manually
    let sum_val: i64 = raw.split("\"sum\":").nth(1).unwrap()
        .split(',').next().unwrap()
        .split('}').next().unwrap()
        .trim().parse().unwrap();
    println!("{}", sum_val);
}
